Platform/OpenNtfsDxe/NTFS.c

```diff
@@ -0,0 +1,376 @@
+/** @file
+  Copyright (c) 2022, Mikhail Krichanov. All rights reserved.
+  SPDX-License-Identifier: BSD-3-Clause
+
+  Functional and structural descriptions follow NTFS Documentation
+  by Richard Russon and Yuval Fledel
+**/
+
+#include "NTFS.h"
+#include "Helper.h"
+
+#define LANGUAGE_CODE_ENGLISH "eng"
+
+CHAR16  *gNTFSDriverName     = L"NTFS Driver";
+
+EFI_COMPONENT_NAME_PROTOCOL
+gNTFSDriverNames = {
+  .GetDriverName      = NTFSCtlDriverName,
+  .GetControllerName  = NTFSCtlGetControllerName,
+  .SupportedLanguages = LANGUAGE_CODE_ENGLISH
+};
+
+EFI_DRIVER_BINDING_PROTOCOL
+gNTFSDriverBinding = {
+  .Supported           = NTFSSupported,
+  .Start               = NTFSStart,
+  .Stop                = NTFSStop,
+  //
+  // UDK/MdeModulePkg/Core/Dxe/Hand/DriverSupport.c
+  // CoreConnectSingleController():589 states
+  // "Sort the remaining DriverBinding Protocol based on their Version field from
+  // highest to lowest." Thus if NTFSVersion > FatVersion, it will somehow
+  // override Fat driver. FatVersion == 0xa.
```

Comment:
This should make sense. We use the latest driver which supports this device.

Therefore, if I'm not wrong, shouldn't we be more correctly identifying whether we in fact DO support each drive (i.e. whether or not it is NTFS) in `NTFSSupported`? (In which case, it would never return 'yes' for FAT drives, and the commented issue would not arise.)

Reply:
Fixed NTFSSupported().